File: Sorting/bubble_sort.c
//C program for sorting the array using bubble sort
#include <stdio.h>
#include <stdlib.h> 

void accept(int a[], int n);
void display(int a[], int n);
void bubble(int a[], int n);

void main()
{
    int n, a[100],i;
    printf("\n Enter the no of elements");
    scanf("%d", &n);
    accept(a, n);
     bubble(a, n);
    display(a, n);
}

void display(int a[], int n)
{
    int i;
    for (i = 0; i < n; i++)
    {
        printf("\t%d", a[i]);
    }
}
void accept(int a[], int n)
{
    int i;
    for (i = 0; i < n; i++)
    {
        scanf("%d",&a[i]);
    }
}

void bubble(int a[], int n)
{
    int i, j, temp;
    for (i = 0; i < n - 1; i++)
    {
        for (j = 0; j < n - i - 1; j++)
        {
            if (a[j] > a[j + 1])
            {
                temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
            }
        }
    }
}

File: Algorithms/Sorting/Sorting_Java/BubbleSort.java
import java.util.Scanner;

class BubbleSort {
    public static int[] bubbleSort (int[] arr) {
        // Temperory varaible for sorting
        int temp;
        int n = arr.length;
        // Bubble sort algorithm
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n - 1; ++j) {
                if (arr[j] > arr[j + 1]) {
                    temp = arr[j + 1];
                    arr[j + 1] = arr[j];
                    arr[j] = temp;
                }
            }
        }

        return arr;
    }

    public static void main(String[] args) {
        int a[];   // only a array reference
        int n;

        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of the array:");
        n = sc.nextInt();

        a = new int[n];

        // Take array input
        System.out.println("Enter values into array:");
        for (int i = 0; i < n; ++i)
        {
            System.out.printf("Enter %d index elem:", i);
            a[i] = sc.nextInt();
        }

        // Apply bubble sort
        a = bubbleSort(a);

        // Print array
        System.out.print("Array: ");
        for (int i = 0; i < n; ++i)
        {
            System.out.print(a[i] + " ");

        }
        System.out.println();
    }
}
